hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/util/UtilityTest.java

```diff
@@ -131,4 +153,81 @@ void instantToTimestamp(long seconds, int nanos) {
     void toSnakeCase(String input, String output) {
         assertThat(Utility.toSnakeCase(input)).isEqualTo(output);
     }
+
+    @ParameterizedTest(name = "Log format {0} with params {1} expecting {2}")
+    @CsvSource(
+            textBlock =
+                    """
+                plain message, , plain message
+                {} arg message, 'one, exception', one arg message
+                {} {} message, 'a, b', a b message
+                {} {} {} {}, 'a, b, c, d, exception', a b c d
+            """)
+    void handleRecoverableErrorLogOrThrow(
+            String format,
+            @ConvertWith(ObjectArrayConverter.class) Object[] args,
+            String formatted,
+            CapturedOutput capturedOutput) {
+
+        var causeProvided = args.length > 0 && args[args.length - 1] instanceof Throwable throwable ? throwable : null;
+
+        /*
+         * With halt on error not set, verify expected log output is generated.
+         */
+        System.setProperty(HALT_ON_ERROR_PROPERTY, "false");
+        if (args.length == 0) {
+            Utility.handleRecoverableError(format);
+        } else {
+            Utility.handleRecoverableError(format, args);
+        }
+
+        var allOutput = capturedOutput.getAll();
+        assertThat(allOutput).contains(RECOVERABLE_ERROR + formatted);
+        if (causeProvided != null) {
+            assertThat(allOutput).contains(causeProvided.getClass().getName() + ": " + causeProvided.getMessage());
+        }
```

Comment:
Is it not possible to just put the `RuntimeException: provided cause` into the expected message so that the entire output sans prefix is in the `@CsvSource`?

```suggestion
        assertThat(capturedOutput.getAll()).contains(RECOVERABLE_ERROR + formatted);
```

Reply:
Indeed. Nice series of refinements. Thank you. Done.